src/SFA.DAS.Payments.DataLocks.DataLockService/DataLockService.cs

```diff
@@ -41,15 +46,34 @@ public class DataLockService : Actor, IDataLockService
             this.paymentLogger = paymentLogger;
             this.apprenticeshipRepository = apprenticeshipRepository;
             this.apprenticeships = apprenticeships;
+            this.learnerMatcher = learnerMatcher;
+            this.courseValidator = courseValidator;
         }
 
         public async Task<DataLockEvent> HandleEarning(ApprenticeshipContractType1EarningEvent message, CancellationToken cancellationToken)
         {
-            var apprenticeshipsForUln = await apprenticeships.TryGet(message.Learner.Uln.ToString(), cancellationToken)
-                .ConfigureAwait(false);
-            var apprenticeship = apprenticeshipsForUln.Value.FirstOrDefault();
+            var learnerMatchResult = await learnerMatcher.MatchLearner(message.Ukprn);
+
+            if (learnerMatchResult.DataLockErrorCode.HasValue)
+            {
+                // TODO: return non-payable earning
+                return null;
```

Comment:
I don't think we should return `null` from a `Task<T>` method, it'll cause a NullReferenceException in the calling `await` without the caller having to do some null checking.  

```suggestion
                return Task.FromResult<DataLockEvent>(null);
```

Reply:
This is only a temporary thing. one of the DataLock component tasks involves publishing NonPayableEvents and at that point, the correct values could be returned.

Adding the suggested code in doesn't compile as the method is async so the result needs to be a DataLockEvent.

Reply:
Ah yes, I missed the `async` on the method.